From a0c0a082eaacb4619a68351702e69bbd4e16d65a Mon Sep 17 00:00:00 2001
From: Gernot Tenchio <gernot.tenchio@securepoint.de>
Date: Wed, 24 Nov 2010 11:28:01 +0100
Subject: [PATCH 2/2] hostapd: fix install path

---
 hostapd/Makefile |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/hostapd/Makefile b/hostapd/Makefile
index 2db0707..8eba33e 100644
--- a/hostapd/Makefile
+++ b/hostapd/Makefile
@@ -757,8 +757,13 @@ verify_config:
 		exit 1; \
 	fi
 
+prefix=/usr/local
+
 install: all
-	for i in $(ALL); do cp -f $$i /usr/local/bin/$$i; done
+	@mkdir -p $(DESTDIR)$(prefix)/bin
+	@mkdir -p $(DESTDIR)$(prefix)/sbin
+	cp -f hostapd $(DESTDIR)$(prefix)/sbin
+	cp -f hostapd_cli $(DESTDIR)$(prefix)/bin
 
 ../src/drivers/build.hostapd:
 	@if [ -f ../src/drivers/build.wpa_supplicant ]; then \
-- 
1.7.2.3

