diff -ruN avifile-0.6.orig/autogen.sh avifile-0.6/autogen.sh
--- avifile-0.6.orig/autogen.sh	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/autogen.sh	2007-09-19 20:07:53.000000000 +0200
@@ -196,9 +196,9 @@
     echo "	 automake 1.4 or better (automake 1.6.0 is buggy!)"
     echo "	 autoconf 2.52 or better"
     echo "Your installed version:"
-    $use_libtool --version | head -1
-    $use_automake --version | head -1
-    $use_autoconf --version | head -1
+    $use_libtool --version | head -n1
+    $use_automake --version | head -n1
+    $use_autoconf --version | head -n1
     echo "Report aclocal = "
     $use_aclocal --print-ac-dir
     echo "Please report your problem on kabi@users.sourceforge.net"
diff -ruN avifile-0.6.orig/config.guess avifile-0.6/config.guess
--- avifile-0.6.orig/config.guess	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/config.guess	2007-09-19 20:08:39.000000000 +0200
@@ -197,7 +197,7 @@
 	echo m68k-sun-sunos${UNAME_RELEASE}
 	exit 0 ;;
     sun*:*:4.2BSD:*)
-	UNAME_RELEASE=`(head -1 /etc/motd | awk '{print substr($5,1,3)}') 2>/dev/null`
+	UNAME_RELEASE=`(head -n1 /etc/motd | awk '{print substr($5,1,3)}') 2>/dev/null`
 	test "x${UNAME_RELEASE}" = "x" && UNAME_RELEASE=3
 	case "`/bin/arch`" in
 	    sun3)
@@ -376,7 +376,7 @@
 	fi
 	exit 0 ;;
     *:AIX:*:4)
-	IBM_CPU_ID=`/usr/sbin/lsdev -C -c processor -S available | head -1 | awk '{ print $1 }'`
+	IBM_CPU_ID=`/usr/sbin/lsdev -C -c processor -S available | head -n1 | awk '{ print $1 }'`
 	if /usr/sbin/lsattr -EHl ${IBM_CPU_ID} | grep POWER >/dev/null 2>&1; then
 		IBM_ARCH=rs6000
 	else
diff -ruN avifile-0.6.orig/libmmxnow/autogen.sh avifile-0.6/libmmxnow/autogen.sh
--- avifile-0.6.orig/libmmxnow/autogen.sh	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/libmmxnow/autogen.sh	2007-09-19 20:10:19.000000000 +0200
@@ -176,9 +176,9 @@
 	echo "   automake 1.4 or better"
 	echo "   autoconf 2.52 or better"
 	echo "Your installed version:"
-        $use_libtool --version | head -1
-        $use_automake --version | head -1
-        $use_autoconf --version | head -1
+        $use_libtool --version | head -n1
+        $use_automake --version | head -n1
+        $use_autoconf --version | head -n1
         echo "Report aclocal = "
         $use_aclocal --print-ac-dir
         echo "Please report your problem on avifile-admin@prak.org"
