diff -ruN avifile-0.6.orig/samples/qtvidcap/kv4lsetup.cpp avifile-0.6/samples/qtvidcap/kv4lsetup.cpp
--- avifile-0.6.orig/samples/qtvidcap/kv4lsetup.cpp	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/samples/qtvidcap/kv4lsetup.cpp	2007-09-19 21:20:17.000000000 +0200
@@ -233,7 +233,7 @@
     if (XF86DGAQueryExtension(dpy, &foo, &bar)) {
 	XF86DGAQueryDirectVideo(dpy, XDefaultScreen(dpy),&flags);
 	if (flags & XF86DGADirectPresent) {
-	    XF86DGAGetVideoLL(dpy,XDefaultScreen(dpy),(int*)&base,&width,&foo,&bar);
+	    XF86DGAGetVideoLL(dpy,XDefaultScreen(dpy),(unsigned int*)&base,&width,&foo,&bar);
 	    set_bpl  = width * set_bpp/8;
 	    set_base = base;
 	    if (verbose == 2) {
diff -ruN avifile-0.6.orig/samples/qtvidcap/v4lxif.cpp avifile-0.6/samples/qtvidcap/v4lxif.cpp
--- avifile-0.6.orig/samples/qtvidcap/v4lxif.cpp	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/samples/qtvidcap/v4lxif.cpp	2007-09-19 21:30:53.000000000 +0200
@@ -1140,7 +1140,7 @@
 		    major, minor, XDGA_MAJOR_VERSION, XDGA_MINOR_VERSION);
 		have_dga = false;
 	    } else {
-		XF86DGAGetVideoLL(disp, DefaultScreen(disp), (int*)&base, &width, &bank, &ram);
+		XF86DGAGetVideoLL(disp, DefaultScreen(disp), (unsigned int*)&base, &width, &bank, &ram);
 		if (!base)
 		    AVMOUT(AVML_WARN, "Can not allocate frame buffer base!");
 	    }
