diff -ruN avifile-0.6.orig/Makefile.am avifile-0.6/Makefile.am
--- avifile-0.6.orig/Makefile.am	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/Makefile.am	2007-09-20 21:12:01.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = . admin m4 include ffmpeg drivers lib plugins bin \
+SUBDIRS = . admin m4 include drivers lib plugins bin \
  doc debian libavqt player samples
 
 EXTRA_DIST = INSTALL autogen.sh avifile.spec.in avifile.pc.in \
diff -ruN avifile-0.6.orig/configure.in avifile-0.6/configure.in
--- avifile-0.6.orig/configure.in	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/configure.in	2007-09-20 21:12:01.000000000 +0200
@@ -1017,25 +1017,10 @@
 AC_SUBST(Z_LIBS)
 
 dnl mpglib/Makefile
-dnl ffmpeg/libavfilter/Makefile
 
 AC_CONFIG_FILES([
  Makefile
  admin/Makefile
- ffmpeg/Makefile
- ffmpeg/libavformat/Makefile
- ffmpeg/libavcodec/Makefile
- ffmpeg/libavcodec/alpha/Makefile
- ffmpeg/libavcodec/armv4l/Makefile
- ffmpeg/libavcodec/i386/Makefile
- ffmpeg/libavcodec/mlib/Makefile
- ffmpeg/libavcodec/ppc/Makefile
- ffmpeg/libavcodec/ps2/Makefile
- ffmpeg/libavcodec/sh4/Makefile
- ffmpeg/libavcodec/sparc/Makefile
- ffmpeg/libavutil/Makefile
- ffmpeg/libpostproc/Makefile
- ffmpeg/libswscale/Makefile
  drivers/Makefile
  drivers/libdha/Makefile
  drivers/libdha/bin/Makefile
diff -ruN avifile-0.6.orig/lib/Makefile.am avifile-0.6/lib/Makefile.am
--- avifile-0.6.orig/lib/Makefile.am	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/lib/Makefile.am	2007-09-20 21:12:01.000000000 +0200
@@ -1,10 +1,7 @@
 DIST_SUBDIRS = aviplay aviread aviwrite common subtitle video
 noinst_HEADERS = Uncompressed.h
 
-COMMONCFLAGS = $(DIVX4_CFLAGS) $(XVID_CFLAGS) -DPLUGIN_PATH=\"$(pkglibdir)\" \
- -I$(srcdir)/../ffmpeg/libavcodec \
- -I$(srcdir)/../ffmpeg/libavformat \
- -I$(srcdir)/../ffmpeg/libavutil
+COMMONCFLAGS = $(DIVX4_CFLAGS) $(XVID_CFLAGS) -DPLUGIN_PATH=\"$(pkglibdir)\" 
 if AMM_USE_FAST_BUILD
 AVISOURCES = libaviplay.cpp
 AVILIBADD =
diff -ruN avifile-0.6.orig/lib/aviread/Makefile.am avifile-0.6/lib/aviread/Makefile.am
--- avifile-0.6.orig/lib/aviread/Makefile.am	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/lib/aviread/Makefile.am	2007-09-20 21:13:58.000000000 +0200
@@ -43,19 +43,7 @@
  StreamInfo.cpp
 
 
-if AMM_LINK_SHARED
-FF_LIB = \
- ../../ffmpeg/libavformat/libaviplayavformat.la \
- ../../ffmpeg/libavcodec/libaviplayavcodec.la \
- ../../ffmpeg/libswscale/libaviplayswscale.la \
- ../../ffmpeg/libavutil/libaviplayavutil.la
-else 
-FF_LIB = \
- ../../ffmpeg/libavformat/libavformat.la \
- ../../ffmpeg/libavcodec/libavcodec.la \
- ../../ffmpeg/libswscale/libswscale.la \
- ../../ffmpeg/libavutil/libavutil.la
-endif
+FF_LIB = 
 
 libaviread_la_LIBADD = $(FF_LIB)
 
diff -ruN avifile-0.6.orig/plugins/Makefile.am avifile-0.6/plugins/Makefile.am
--- avifile-0.6.orig/plugins/Makefile.am	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/plugins/Makefile.am	2007-09-20 21:12:01.000000000 +0200
@@ -20,7 +20,7 @@
 
 AM_CPPFLAGS = -Ilibac3pass -Ilibwin32 -Ilibwin32/audiodec \
  -Ilibwin32/videocodec -Ilibwin32/loader \
- -Ilibwin32/loader/dshow -I../ffmpeg/libavcodec \
+ -Ilibwin32/loader/dshow \
  -DWIN32_PATH=\"$(WIN32_PATH)\" -D__WINE__ -DUNICODE \
  -Ddbg_printf=__vprintf -DTRACE=__vprintf \
  -Wmissing-prototypes -Wimplicit-function-declaration \
diff -ruN avifile-0.6.orig/plugins/libffmpeg/Makefile.am avifile-0.6/plugins/libffmpeg/Makefile.am
--- avifile-0.6.orig/plugins/libffmpeg/Makefile.am	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/plugins/libffmpeg/Makefile.am	2007-09-20 21:12:01.000000000 +0200
@@ -20,11 +20,7 @@
 #ffmpeg_la_LIBADD = ../../ffmpeg/libavcodec/libavcodec.a
 ffmpeg_la_LDFLAGS = -module -avoid-version
 
-AM_CPPFLAGS = $(LTNOPIC) $(INSTRUMENT) \
- -I$(srcdir)/../../ffmpeg/libavcodec \
- -I$(srcdir)/../../ffmpeg/libavformat \
- -I$(srcdir)/../../ffmpeg/libavutil \
- -I$(srcdir)/../../ffmpeg
+AM_CPPFLAGS = $(LTNOPIC) $(INSTRUMENT) 
 AM_CXXFLAGS = $(CXXRTTIEXCEPT)
 
 TEST_LIBS = ../../lib/libaviplay.la $(pkglib_LTLIBRARIES) $(SDL_LIBS)
