diff -ruN avifile-0.6.orig/m4/xvid4.m4 avifile-0.6/m4/xvid4.m4
--- avifile-0.6.orig/m4/xvid4.m4	2000-01-01 00:00:00.000000000 +0100
+++ avifile-0.6/m4/xvid4.m4	2007-09-19 20:03:26.000000000 +0200
@@ -35,7 +35,7 @@
 dnl Now check if the installed XviD is sufficiently new.
 dnl
     AC_AVM_CHECK_LIB(xvid4, [for xvid_global in -lxvidcore (4)],
-                     XVID4_CFLAGS, XVID4_LIBS, -lxvidcore, [
+                     XVID4_CFLAGS, XVID4_LIBS, -lxvidcore -lm, [
 #include <stdio.h>
 #include <string.h>
 #include <xvid.h>], [
