
http://bugzilla.gnome.org/show_bug.cgi?id=462238

diff -ru brasero-0.6.1.orig/src/scsi/scsi-command.c brasero-0.6.1/src/scsi/scsi-command.c
--- brasero-0.6.1.orig/src/scsi/scsi-command.c	2007-08-25 15:16:23.000000000 +0200
+++ brasero-0.6.1/src/scsi/scsi-command.c	2007-09-07 11:49:05.000000000 +0200
@@ -85,7 +85,7 @@
 	memset (transport, 0, sizeof (struct sg_io_hdr));
 	
 	transport->interface_id = 'S';				/* mandatory */
-//	transport->flags = SG_FLAG_LUN_INHIBIT|SG_FLAG_DIRECT_IO;
+//	transport->flags = SG_FLAG_UNUSED_LUN_INHIBIT|SG_FLAG_DIRECT_IO;
 	transport->cmdp = cmd->cmd;
 	transport->cmd_len = cmd->info->size;
 	transport->dxferp = buffer;
diff -ru brasero-0.6.1.orig/src/scsi/scsi-sg.c brasero-0.6.1/src/scsi/scsi-sg.c
--- brasero-0.6.1.orig/src/scsi/scsi-sg.c	2007-08-25 15:16:23.000000000 +0200
+++ brasero-0.6.1/src/scsi/scsi-sg.c	2007-09-07 11:49:32.000000000 +0200
@@ -44,7 +44,7 @@
 	memset (&sense_buffer, 0, sizeof (sense_buffer));
 
 	command->interface_id = 'S';				/* mandatory */
-	command->flags = SG_FLAG_LUN_INHIBIT|SG_FLAG_DIRECT_IO;
+	command->flags = SG_FLAG_UNUSED_LUN_INHIBIT|SG_FLAG_DIRECT_IO;
 
 	/* where to output the scsi sense buffer */
 	command->sbp = sense_buffer;
