--- nut-2.2.0/drivers/Makefile.am.orig	2007-10-16 16:52:04.000000000 +0300
+++ nut-2.2.0/drivers/Makefile.am	2007-10-16 16:52:14.000000000 +0300
@@ -19,7 +19,7 @@
  bestuferrups bestups cpsups cyberpower dummy-ups etapro everups 	\
  gamatronic genericups isbmex liebert masterguard megatec metasys 	\
  mge-shut mge-utalk newmge-shut	nitram oneac optiups powercom rhino 	\
- safenet skel solis tripplite tripplitesu upscode2 victronups powerpanel
+ safenet solis tripplite tripplitesu upscode2 victronups powerpanel
 SNMP_DRIVERLIST = snmp-ups
 USB_LIBUSB_DRIVERLIST = usbhid-ups bcmxcp_usb tripplite_usb megatec_usb
 USB_HIDDEV_DRIVERLIST = energizerups
